osgp/platform/osgp-domain-core/src/main/java/org/opensmartgridplatform/domain/core/validation/LongTermIntervalAndLongTermIntervalTypeValidator.java

```diff
@@ -23,39 +23,39 @@ public void initialize(final LongTermIntervalAndLongTermIntervalType constraintA
 
     @Override
     public boolean isValid(final Configuration value, final ConstraintValidatorContext context) {
-        if (value == null) {
+        if (value == null)
             return true;
+
+        // LongTermHistoryInterval and LongTermHistoryIntervalType must
+        // either be: BOTH PRESENT or BOTH NOT PRESENT
+        // Using the exclusive or operator this becomes
+        if (value.getLongTermHistoryInterval() == null ^ value.getLongTermHistoryIntervalType() == null)
+            return false;
+
+        // And, the value of long term history interval must be among the
+        // permitted values
+        // First check upper limit depending on interval type:
+        // - maximal 30 for long term history interval type DAYS.
+        // - maximal 12 for long term history interval type MONTHS.
+        switch (value.getLongTermHistoryIntervalType()) {
+        case DAYS:
+            if (value.getLongTermHistoryInterval() > 30)
+                return false;
+            break;
+        case MONTHS:
+            if (value.getLongTermHistoryInterval() > 12)
+                return false;
+            break;
+        default:
+            throw new ValidationException("unknown LongTermHistoryIntervalType");
         }
-        // If LongTermHistoryInterval or LongTermHistoryIntervalType is not
-        // present,
-        // the other must be not present as well.
-        if (value.getLongTermHistoryInterval() == null && value.getLongTermHistoryIntervalType() == null) {
-            return true;
-        }
-        // If LongTermHistoryInterval or LongTermHistoryIntervalType is present,
-        // the other must be as well.
-        if (value.getLongTermHistoryInterval() != null && value.getLongTermHistoryIntervalType() != null) {
-            // And, the value of long term history interval must be among the
-            // permitted values, defined by the ranges:
-            // - from 1 to 30 for long term history interval type DAYS.
-            // - from 1 to 12 for long term history interval type MONTHS.
-            switch (value.getLongTermHistoryIntervalType()) {
-            case DAYS:
-                if (value.getLongTermHistoryInterval() >= 1 && value.getLongTermHistoryInterval() <= 30) {
-                    return true;
-                }
-                break;
-            case MONTHS:
-                if (value.getLongTermHistoryInterval() >= 1 && value.getLongTermHistoryInterval() <= 12) {
-                    return true;
-                }
-                break;
-            default:
-                throw new ValidationException("unknown LongTermHistoryIntervalType");
-
-            }
-        }
 
-        return false;
+        // Check lower limit:
+        // - minimal 1
+        if (value.getLongTermHistoryInterval() < 1)
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Replace this if-then-else statement by a single return statement. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://ci.opensmartgridplatform.org/sonarqube/coding_rules#rule_key=squid%3AS1126)